api/src/main/java/org/openmrs/module/attachments/AttachmentsService.java

```diff
@@ -11,13 +12,48 @@
  * Used for managing attachments
  */
 public interface AttachmentsService {
-	
+
+
+	/**
+	 * Get all attachments for given patient.
+	 *
+	 * @param patient the patient whose attachment to get
+	 * @param includeRetired
+	 * @return matching attachments
+	 * @throws APIException
+	 */
 	List<Attachment> getAttachments(Patient patient, boolean includeRetired);
-	
+
+	/**
+	 * Get all attachments for given encounter.
+	 *
+	 * @param patient the patient whose attachment to get
+	 * @param encounter the encounter which attachment to get
+	 * @param includeRetired
+	 * @return matching list of attachments
+	 * @throws APIException
+	 */
 	List<Attachment> getAttachments(Patient patient, Encounter encounter, boolean includeRetired);
-	
+
+	/**
+	 * Get all attachments for given visit.
+	 *
+	 * @param patient the patient whose attachments to get
+	 * @param visit the visit which attachments to get
+	 * @param includeRetired
+	 * @return matching list of attachments
+	 * @throws APIException
+	 */
 	List<Attachment> getAttachments(Patient patient, Visit visit, boolean includeRetired);
-	
+
+	/**
+	 * Get all attachments for given patient which are not associated with any encounters.
```

Comment:
→ `Get a patient's attachments that are not associated with any visits or encounters.`